mechanicalsoup/browser.py

```diff
@@ -180,11 +180,12 @@ def _request(self, form, url=None, **kwargs):
                     # in browsers, file upload only happens if the form
                     # (or submit button) enctype attribute is set to
                     # "multipart/form-data". We don't care, simplify.
-                    if not value:
-                        continue
-                    if isinstance(value, string_types):
+                    filename = value
+                    if value != "" and isinstance(value, string_types):
```

Comment:
I know I wrote it out this way myself, but when freshly reviewing this statement, the overloaded use of `value` really confused me! Perhaps something like this instead:
```python
filename = value
if filename != "" and isinstance(filename, string_types):
    value = open(filename, "rb")
```
Or to be much more explicit at the cost of a couple more lines of code:
```python
filename = value
if filename != "" and isinstance(filename, string_types):
    content = open(filename, "rb")
else:
    content = ""
files[name] = (filename, content)
```
Or something similar, if you have a better way to write it -- just so that it's more clear that we're working with a "filename" :)

Reply:
Understood and agreed !
I'd side with your second suggestion : more explicit is always nice to avoid unnecessary headaches in the future !